### DISCUSSION TITLE
Cercasi sviluppatore per sviluppare un termostato dedicato ad impianti con pompa di calore

### DISCUSSION THREAD

**Discussion by Bergoglio** [indomus/forum#2964] [HUB: Home Assistant]
Ciao ragazzi,

sono mesi che cerco un programmatore Python decente (anche con contributo economico) per sviluppare un termostato dedicato ad impianti con pompa di calore da far girare in Home Assistant e rilasciare in Open Source. Purtroppo ho trovato nessuno......
Le mie capacità di scrivere codice sono nulle, ho le conoscenze termotecniche per definire le funzioni di questo termostato e le formule di gestione dei parametri ma non riesco a tradurle in software.
Qualcuno ha voglia di cimentarsi con me in questo mini progetto ?

**Comment by M0ntyC0r4** [comment]
penso che tu debba cercare su fiver...so che li questo tipo di professionisti si trovano.

**Reply by Bergoglio** [reply]
Mah, ho provato più volte a guardare questi siti di freelancers e non riesco a districarmi (ammetto i miei limiti) : non riesco mai a trovare una persona che sia vicina a quel che cerco.

**Reply by M0ntyC0r4** [reply]
Sul "come", ahimè, non saprei aiutarti perché da buon boomer, io di questi siti ho solo sentito parlare sul web...

Però so che i cciovani millennials oggi si freelanciano su questi siti qui...e quello mi pare sia molto in voga!!

**Reply by Bergoglio** [reply]
Io sono un poco dopo (generazione X) ma fatico non poco comunque !
Ho guardato questi siti ma nelle descrizioni trovo solo sviluppatori di siti, di software per estrazione / recupero dati o revisori di codice già scritto per ricercare bug. Ma di persone che si offrono di sviluppare propriamente il "ferro" manco l'ombra.
Inoltre ho messo anche alcuni annunci in siti specializzati e non ti immagini che folla di "scienziati del suono" ho dovuto falciare : tutti che conoscevano Home Assistant, tutti che avevano sviluppato dei plugin per installazioni specialistiche di Home Assistant e tutti che tengono dei corsi nelle università (inesistenti) e loro hanno magari il diploma da ragioniere.
Sono convinto che un buon programmatore non va su questi siti perché ad oggi non ha problemi a trovare lavoro e per questo ho provato qui : la speranza è di trovare un appassionato che si voglia cimentare nel tempo libero.

**Reply by M0ntyC0r4** [reply]
in realtà anche io penso di essere dopo, perché gli anni 80.....però mi sento vecchio inside.
Riguardo al resto, che dire, schettino teneva dei corsi all'università sulla gestione del panico....magari i corsi li tenevano davvero!! XD

**Comment by lux90ita** [comment]
in realtà non ho ben capito cosa ti serve :D
dalle pompe di calore hai già dei dati?
ti serve solo la veste grafica del termostato su HA ?
se ti va di condividere in maniera dettagliata sapendo anche i dati che hai già

**Reply by Bergoglio** [reply]
Mah, l'idea era di sviluppare un termostato con logiche di funzionamento adeguate alle pompe di calore (mantenimento della temperatura con erogazione continua di poco calore invece che sbalzi forti addirittura di tipo on/off, come concetto generale); il problema è che non sono in grado di programmare e quindi tutta la mia elaborazione sulle logiche / formule / parametri rimane lettera morta.
Ho però abbandonato il progetto perché il tutto si scontra con un problema tecnico : non esistono componenti finali (termosifoni / riscaldamento a pavimento / ventilconvettori etc.) dimensionati per erogare pochi watt. Quindi l'unica soluzione praticabile sono classici termostati (hardware o creati in Home Assistant) che comandano classici corpi scaldanti che prelevano, per le pompe di calore, da puffer tenuti in temperatura.
